evasquez26/angular4-startup
# S1 one 91 Notifications button

## Story / Task

Here goes the link and info of the task/story.  
Example:

see [ONE-91](https://bixlabs.atlassian.net/browse/ONE-91)


## What this PR does?

Adds the notification button at the top of the navbar.


### Before 

- We did not have notifications button

### After

- We added a notifications button that will display the list of notifications a specific user has
- The notification button will show a list of notifications (max 10)
- If there are more than 10 notifications a "show more" option will show up
- The user can also view all of his notifications clicking the "view all" button at the bottom

### How to test / reproduce

- We need to mock this notifications API
- We need a `db.json` file with a list of notifications with the following structure
```
{
    "notifications": [
    {
      "id": 1,
      "description": "First notification",
      "date": "2017-01-01T10:25:43.511Z",
      "read": true
    }
}
```
- You can add more if you want
- A `routes.json` file with the following
```
{
  "/notifications/user/:id": "/notifications"
}
```
- Then execute `json-server db.json --routes routes.json`
- This will redirect incoming requests from `/notifications/user/:id` to default notifications endpoint
- Open up the list and see that the notifications icon dissapears (it will appear after reload, but the state of the notifications should be saved into the API we assume)
- Observe that there's a list of notifications
- The notifications that are already read should display a check button at the far right
- Notifications that are not read should display a button which the user can click to read the notification

### TODOs
- [ ] Integrate with the real notification API
- [ ] Apply styles
- [ ] Redirect to corresponding section

### Related PRs

N/A

## Migrations

NO

Thanks!
